zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/KerberosInterpreter.java

```diff
@@ -48,23 +48,38 @@ public KerberosInterpreter(Properties property) {
   @ZeppelinApi
   protected abstract boolean runKerberosLogin();
 
-  public String getKerberosRefreshInterval() {
+  @ZeppelinApi
+  protected abstract boolean isKerboseEnabled();
+
+  public void open() {
+    if (isKerboseEnabled()) {
+      startKerberosLoginThread();
+    }
+  }
+
+  public void close() {
+    if (isKerboseEnabled()) {
+      shutdownExecutorService();
+    }
+  }
+
+  private String getKerberosRefreshInterval() {
     if (System.getenv("KERBEROS_REFRESH_INTERVAL") == null) {
```

Comment:
I think we need to add javadoc for `KerberosInterpreter.java` to explain the internal mechnism of it.  So basically I believe user need to do 3 things to extend `KerberosInterpreter`.

* implement `runKerberosLogin `
* implement `isKerboseEnabled `
* define `KERBEROS_REFRESH_INTERVAL` in interpreter setting. Maybe we could also add abstract method like `getKerborseRefreshInterval` to enforce its sub class to implement it.